### PR TITLE
DEV: Add test for arrow keys navigation in the experimental user menu

### PR DESCRIPTION
**Pull request by OsamaSayegh**
Extracted from https://github.com/discourse/discourse/pull/17379.

### CODE REVIEW

File: app/assets/javascripts/discourse/tests/integration/components/site-header-test.js [hunk 1]
```diff
@@ -135,4 +141,47 @@ module("Integration | Component | site-header", function (hooks) {
...
+
+    await triggerKeyEvent(document, "keydown", "ArrowDown");
+    await triggerKeyEvent(document, "keydown", "ArrowDown");
+    await triggerKeyEvent(document, "keydown", "ArrowDown");
+    await triggerKeyEvent(document, "keydown", "ArrowDown");
+    await triggerKeyEvent(document, "keydown", "ArrowDown");
+
+    focusedTab = document.activeElement;
+    assert.ok(
+      focusedTab.href.endsWith("/u/eviltrout/preferences"),
```
Comment (tgxworld) [inline app/assets/javascripts/discourse/tests/integration/components/site-header-test.js:168]: Just curious, is there no id or class we can use this identify the element here?

**Review by tgxworld — APPROVED** [review]
Aside from a comment, code looks good.

File: app/assets/javascripts/discourse/tests/integration/components/site-header-test.js (hunk 1 above)
Comment (OsamaSayegh) [reply app/assets/javascripts/discourse/tests/integration/components/site-header-test.js:168]: Yeah there isn't any unique identifier for this element, but we will replace this element/link with a button soon and it'll have an ID then and I will update this test.

**Comment by OsamaSayegh** [comment]
thank you tgx!

**Pull request merged**